apps/web/src/pages/quick-start/steps/Setup.tsx

```diff
@@ -146,8 +148,14 @@ const LoaderWrapper = styled.div`
   margin-top: 10px;
 `;
 
-function updateCodeSnipped(codeSnippet: string, environmentIdentifier: string) {
-  return codeSnippet.replace(APPLICATION_IDENTIFIER, environmentIdentifier);
+function updateCodeSnipped(codeSnippet: string, environmentIdentifier: string, apiKey: string) {
+  const concatUrls = process.env.REACT_APP_ENVIRONMENT !== 'prod' || !!process.env.REACT_APP_DOCKER_HOSTED_ENV;
+
+  return codeSnippet
+    .replace(APPLICATION_IDENTIFIER, environmentIdentifier)
+    .replace(API_KEY, apiKey ?? '')
+    .replace(BACKEND_API_URL, concatUrls ? process.env.REACT_APP_API_URL || 'http://localhost:3000' : '')
+    .replace(BACKEND_SOCKET_URL, concatUrls ? process.env.REACT_APP_WS_URL || 'http://localhost:3002' : '');
```

Comment:
Shouldn't we use `window._env_.REACT_APP_API_URL` and `window._env_.REACT_APP_WS_URL` to avoid to have spread all the Web app the URL calculation?

https://github.com/novuhq/novu/blob/6e5b03ef3fadca0121957d1dd4eab13e863831e9/apps/web/src/config/index.ts#L12

If for any reason they are not the right variables to use here, maybe centralising in the `config` the calculation would be a good idea to have all that isolated in one place and not through the codebase. What do you think?

Reply:
Good catch, I will add it to the config file. Thanks!